cli/index.js

```diff
@@ -246,45 +245,41 @@ if (!paths.length) {
 
 /** Given an array of "things" return all distinct .json files.
  *
- * Note that these "things" can be a directory, a file path, or a
- * pattern.
+ * Note that these "things" can be a directory, a file path, or a list of
+ * file paths.
  * Only if each thing is a directory do we search for *.json files
  * in there recursively.
  */
 function expandFiles(directoriesPatternsOrFiles) {
-  function findFiles(directory) {
+  function findFiles(directory, extension = ".json", filepaths = []) {
     if (path.basename(directory) === "node_modules") {
       throw new Error(
         `Can't dig deeper into ${directory}. ` +
           `Doesn't look like stumptown content packaged location`
       );
     }
-    const found = glob.sync(path.join(directory, "*.json"));
-
-    fs.readdirSync(directory, { withFileTypes: true })
-      .filter(dirent => dirent.isDirectory())
-      .map(dirent => path.join(directory, dirent.name))
-      .map(findFiles)
-      .forEach(files => found.push(...files));
-
-    return found;
+    const files = fs.readdirSync(directory);
```

Comment:
bit-of-a-nit: instead of mutating parameters and nested functions, I'd prefer something like:

pseudo-ish code:
```javascript
function expandFiles(paths) {
  const allPaths = paths.slice();
  const files = [];
  for (const path of allPaths) {
    if (path is JSON file) {
      files.push(path);
    } else if (path is directory) {
      allPaths.concat(readdirSync(path));
    }
  }

  return files;
}
```

Reply:
I don't get it. Where's my code mutating? 
Also, where's the recursion in your pseudo-code? It needs to dig deeper into directories.

Reply:
`filepaths.push(filepath)` mutates the given filepath array, doesn't it? My pseudo-code above would also dig deeper by adding found files to the `allPaths` array. Happy to create a PR with real code later to make it clearer what I mean.

Reply:
Mind you, not all mutations are bad. The ones that lead to tricky bugs where the order, of doing things, was not supposed to matter, or React where you might accidentally mix mutating `this.state` with calls to `this.setState`. 
I'm still looking forward to see a followup here. Stumptown is and might be about looping of thousands of files so perf and max. memory usage might actually matter.